### PR TITLE
974 improve logger

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

**Comment by gregLibert** [comment]
This PR is to be reviewed by commits:
The second one (351471d19287b777c166059bb8625798b284d58b) is huge, but it's functionaly dumb simple.

**Review by thomas-senechal — APPROVED** [review]
Great improvements ! Looks much cleaner

**Review by Pierregvx — APPROVED** [review]

File: pkg/logger/global.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package logger
+
+// Initialize the logger at package level.
+var (
+	//nolint:gochecknoglobals
+	logger *Logger
+)
+
+// InitializeGlobal initializes the global logger at package level.
+// Once initialized, the logger can be used by calling the directly the classic logging functions (Debug, Info...).
```
Comment (Thykof) [inline pkg/logger/global.go:10]:
```suggestion
// Once initialized, the logger can be used by callin directly the classic logging functions (Debug, Info...).
```

File: pkg/logger/global.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+// Note: The global logger must be initialized before calling this function, otherwise it will panic.
+func Debug(args ...interface{}) {
+	if logger == nil {
+		panic("global logger not initialized")
+	}
+
+	logger.Debug(args...)
+}
+
+// Debug logs a message at the debug level.
```
Comment (Thykof) [inline pkg/logger/global.go:30]:
```suggestion
// Debugf logs a message at the debug level.
```

File: pkg/logger/global.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+	return err
+}
+
+// Debug logs a message at the debug level.
+// This function takes a variable number of arguments and uses fmt.Sprint to construct the log message.
+// Note: The global logger must be initialized before calling this function, otherwise it will panic.
+func Debug(args ...interface{}) {
+	if logger == nil {
+		panic("global logger not initialized")
```
Comment (Thykof) [inline pkg/logger/global.go:24]: this string is repeated many time, let's put it into a constant variable at the beginning of the file.

File: pkg/logger/level_test.go
Comment (Thykof) [inline pkg/logger/level_test.go:1]: we use package "github.com/stretchr/testify/assert" to assert in tests.

**Review by Thykof — CHANGES_REQUESTED** [review]

File: cmd/massastation/main.go [hunk 4]
```diff
@@ -56,23 +58,27 @@ func ParseFlags() api.StartServerFlags {
...
+	if err != nil {
+		log.Fatalf("while initializing logger: %s", err.Error())
+	}
+
+	defer logger.Close()
 
 	flags := ParseFlags()
 	if flags.Version {
-		logger.Logger.Infof("Version:%s", config.Version)
+		logger.Infof("Version:%s", config.Version)
```
Comment (Pierregvx) [inline cmd/massastation/main.go:70]: I think we should use the regular log for this one : 
```
$ massastation --version
2023-07-07T17:22:21.790+0200	INFO	massastation/main.go:64	Version:v0.2.5

```

Comment (Thykof) [reply cmd/massastation/main.go:70]: what is the regular log?

**Review by Thykof — APPROVED** [review]
nice!

File: cmd/massastation/main.go (hunk 4 above)
Comment (gregLibert) [reply cmd/massastation/main.go]: I simply printed it. I also fixed a linter warning.

File: pkg/logger/global.go (hunk 1 above)
Comment (gregLibert) [reply pkg/logger/global.go]: should be fixed

File: pkg/logger/global.go (hunk 2 above)
Comment (gregLibert) [reply pkg/logger/global.go]: should be fixed

File: pkg/logger/global.go (hunk 3 above)
Comment (gregLibert) [reply pkg/logger/global.go]: should be fixed

File: pkg/logger/level_test.go
Comment (gregLibert) [reply pkg/logger/level_test.go:1]: should be fixed

**Comment by SlnPons** [comment]
Can I test this issue myself? 
Happy to trust you guys if you tested it

**Comment by Thykof** [comment]
For a developer point of view I think it's a bit shame that we don't see in the log the correct location in the code:

`2023-07-12T09:26:45.171+0200	INFO	logger/global.go:62	HTTP server Shutdown: context deadline exceeded`

We have `logger/global.go:62` but before we had the exact location where the log function is called. 

However we don't really need this information.

**Pull request merged**